subiquity/server/controllers/install.py

```diff
@@ -259,7 +259,14 @@ async def run_curtin_step(name, stages, step_config, source=None):
 
         await run_curtin_step(name="initial", stages=[], step_config={})
 
-        if fs_controller.is_core_boot_classic():
+        if fs_controller.reset_partition_only:
```

Comment:
This method is getting a bit long.  I suggest we start splitting this method up, maybe by creating an install type and doing single dispatch stuff.  Doesn't have to be this PR.

Reply:
yeah, it's a long series of involved function calls that are very similar but not quite the same